### PR TITLE
[ENH] Update registration subworkflows and modules

### PR DESCRIPTION
**Pull request by AlexVCaron**
Continuation of #25

## Type of improvement

- [ ] Documentation
- [ ] Development tools (e.g. linter, formatter, etc.)
- [ ] Development container
- [ ] Global update (please specify)
- [x] Other (please specify)

## Describe your improvement

subworkflow/registration:

- add synthregistration optionality to subworkflow
- add specific test for synthregistration

modules/synthregistration:

- change output structure to tuple transfo_image containing both transforms
- add python package access. 

module/convert:

- add optionality to affine conversion. If no affine is provided, skips conversion. Synthmorph output include affine initialization in its deform output.
- change output order

## Checklist before requesting a review

- [x] Ensure the syntax is correct (**EditorConfig** and **Prettier** must pass)
- [x] Run the test suites if your changes affect any module
- [x] Regenerate the **Poetry** lock file if you have updated the dependencies
- [x] Ensure the documentation is up-to-date

### CODE REVIEW

File: subworkflows/nf-neuro/registration/tests/main.nf.test [hunk 1]
```diff
@@ -155,4 +156,46 @@ nextflow_workflow {
...
+    test("registration - synthregistration") {
+        config "./nextflow_synthregistration.config"
+        when {
+            workflow {
+                """
+                ch_split_test_data = LOAD_DATA.out.test_data_directory
+                    .branch{
+                        t1w: it.simpleName == "T1w"
+                        b0: it.simpleName == "b0"
+                        dti: it.simpleName == "dti"
```
Comment (arnaudbore) [inline subworkflows/nf-neuro/registration/tests/main.nf.test:168]: don't need it here.

File: modules/nf-neuro/registration/synthregistration/main.nf [hunk 2]
```diff
@@ -52,12 +51,12 @@ process REGISTRATION_SYNTHREGISTRATION {
...
 
     touch ${prefix}__output_warped.nii.gz
-    touch ${prefix}__affine_warp.lta
     touch ${prefix}__deform_warp.nii.gz
+    touch ${prefix}__affine_warp.lta
 
     cat <<-END_VERSIONS > versions.yml
     "${task.process}":
-        Freesurfer: 7.4
+        Freesurfer: 7.4.1
```
Comment (arnaudbore) [inline modules/nf-neuro/registration/synthregistration/main.nf:59]: why do you have different soft and versions here ? Generic way of getting the version also.

**Review by arnaudbore — APPROVED** [review]
Quick questions and GTG !

File: modules/nf-neuro/registration/synthregistration/main.nf (hunk 2 above)
Comment (AlexVCaron) [reply modules/nf-neuro/registration/synthregistration/main.nf:59]: Not for synthmorph yet, no mechanism in the container sadly. I aligned both versions for now

**Comment by AlexVCaron** [comment]
Let's wait a bit. I found an inconsistency with the last two conditions in the subworkflow. THere should not be a check on `ch_metric` directly, it's bound to fail as we said. I rewrote the section so it's channels doing the job of selecting either `anatotodwi` or the classical `ants` SyN script.

**Review by arnaudbore — APPROVED** [review]

**Pull request merged**